src/frontend/js/events/conditional-fields.ts

```diff
@@ -76,4 +76,18 @@ window.addEventListener("load", () => {
       );
     });
   }
+
+  // event handler to show location selection
+  const locationCheckbox = document.getElementById(
+    "id_has_not_location"
+  );
+  if (locationCheckbox) {
+    locationCheckbox.addEventListener("click", () => {
+      document
+        .getElementById("location-block")
+        .classList.toggle("hidden");
+      let locationInput = document.getElementById("id_location") as HTMLInputElement;
+      locationInput.disabled = !locationInput.disabled;
+    });
+  }
```

Comment:
```suggestion
  document
    .getElementById("id_has_not_location")
    ?.addEventListener("click", () => {
      document
        .getElementById("location-block")
        .classList.toggle("hidden");
      const locationInput = document.getElementById("id_location") as HTMLInputElement;
      locationInput.disabled = !locationInput.disabled;
    });
```
I think with TypeScript, we can simplify this a little bit.

Reply:
I just googled that syntax, its called the Elvis operator. Best name for an operator so far :smile:

Reply:
Haha, indeed that's a good name :laughing: 
However, in this context, `?.` is not the Elvis operator (which returns the first part if it evaluates to true), but rather reads as "continue with this evaluation if it's true until here", see documentation about [Optional Chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html)...
This way, we don't have to check explicitly, whether an element exists, but can just do something if it exists and ignore the code otherwise.

Reply:
Its a handy operator, I will keep it in mind :+1: